pkg/driver/bacnet/merge/poll_test.go

```diff
@@ -0,0 +1,163 @@
+			name: "no deadline - test succeeds after default deadline",
+			args: args{
+				ctx: func() context.Context {
+					return context.Background()
+				},
+				timeout: 10 * time.Millisecond,
+				test: func() func(t B) bool {
+					count := 0
+					return func(t B) bool {
+						count++
+
+						// fail until at least 10 ms has passed
+						if count > 2 {
+							return true
+						}
+
+						return false
+					}
+				},
+			},
+			wantErr: context.DeadlineExceeded,
+		},
+		{
+			name: "no deadline - test succeeds before default deadline",
+			args: args{
+				ctx: func() context.Context {
+					return context.Background()
+				},
+				timeout: 10 * time.Millisecond,
+				test: func() func(t B) bool {
+					return func(t B) bool {
+						// succeed straight away
+						return true
+					}
+				},
+			},
+			wantErr: nil,
+			want:    true,
+		},
+		{
+			name: "deadline before timeout - test succeeds after deadline",
+			args: args{
+				ctx: func() context.Context {
+					ctx, _ := context.WithTimeout(context.Background(), 5*time.Millisecond)
+					return ctx
+				},
+				timeout: 10 * time.Millisecond,
+				test: func() func(t B) bool {
+					count := 0
+					return func(t B) bool {
+						count++
+						if count > 2 {
+							return true
+						}
+						return false
+					}
+				},
+			},
+			wantErr: context.DeadlineExceeded,
+			want:    false,
+		},
+		{
+			name: "deadline before timeout - test succeeds before deadline",
+			args: args{
+				ctx: func() context.Context {
+					ctx, _ := context.WithTimeout(context.Background(), 5*time.Millisecond)
+					return ctx
+				},
+				timeout: 10 * time.Millisecond,
+				test: func() func(t B) bool {
+					return func(t B) bool {
+						return true
+					}
+				},
+			},
+			wantErr: nil,
+			want:    true,
+		},
+		{
+			name: "deadline after timeout - test succeeds after deadline",
+			args: args{
+				ctx: func() context.Context {
+					ctx, _ := context.WithTimeout(context.Background(), 50*time.Millisecond)
+					return ctx
+				},
+				timeout: 10 * time.Millisecond,
+				test: func() func(t B) bool {
+					count := 0
+					return func(t B) bool {
+						count++
+						if count > 2 {
+							return true
+						}
+						return false
+					}
+				},
+			},
+			wantErr: context.DeadlineExceeded,
+			want:    false,
+		},
+		{
+			name: "deadline after timeout - test succeeds before deadline",
+			args: args{
+				ctx: func() context.Context {
+					ctx, _ := context.WithTimeout(context.Background(), 50*time.Millisecond)
+					return ctx
+				},
+				timeout: 10 * time.Millisecond,
+				test: func() func(t B) bool {
+					return func(t B) bool {
+						return true
+					}
+				},
+			},
+			wantErr: nil,
+			want:    true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			res, err := pollUntil[B](tt.args.ctx(), tt.args.timeout, poll, tt.args.test())
+
+			if diff := cmp.Diff(res, tt.want); diff != "" {
+				t.Errorf("pollUntil(): (-got +want)\n%s", diff)
```

Comment:
Again, sorry I didn't spot this earlier. Standard usage is `cmp.Diff(want, got)` with the message saying `(-want,+got)`